elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -782,56 +781,98 @@ public boolean isIdGenerated() {
+     *
+     * @param relation relation name
+     * @param id single id to lookup
+     * @return The PersistentResource of the sought id or null if does not exist.
+     */
+    public PersistentResource getRelation(String relation, String id) {
+        Set<PersistentResource> resources = getRelation(relation, Collections.singletonList(id),
+                Optional.empty(), Optional.empty(), Optional.empty());
+      if (resources.isEmpty()) {
+            return null;
+      }
+      return resources.iterator().next();
+    }
+
+    /**
      * Load a single entity relation from the PersistentResource. Example: GET /book/2
      *
      * @param relation the relation
-     * @param id the id
+     * @param ids the id
      * @return PersistentResource relation
      */
-    public PersistentResource getRelation(String relation, String id) {
+    public Set<PersistentResource> getRelation(String relation,
+                                          List<String> ids,
+                                          Optional<FilterExpression> filter,
+                                          Optional<Sorting> sorting,
+                                          Optional<Pagination> pagination) {
 
-        Optional<FilterExpression> filterExpression;
+        FilterExpression filterExpression = null;
         boolean skipNew = false;
 
         Class<?> entityType = dictionary.getParameterizedType(getResourceClass(), relation);
 
+        Set<PersistentResource> relations = new LinkedHashSet<>();
+
         /* If this is a bulk edit request and the ID we are fetching for is newly created... */
         if (entityType == null) {
             throw new InvalidAttributeException(relation, type);
-        } else if (requestScope.isMutatingMultipleEntities()
-                && requestScope.getObjectById(dictionary.getJsonAliasFor(entityType), id) != null) {
-            filterExpression = Optional.empty();
-            // NOTE: We can safely _skip_ tests here since we are only skipping READ checks on
-            // NEWLY created objects. We assume a user can READ their object in the midst of creation.
-            // Imposing a constraint to the contrary-- at this moment-- seems arbitrary and does not
-            // reflect reality (i.e. if a user is creating an object with values, he/she knows those values
-            // already).
-            skipNew = true;
         } else {
             Class<?> idType = dictionary.getIdType(entityType);
-            Object idVal = CoerceUtil.coerce(id, idType);
             String idField = dictionary.getIdFieldName(entityType);
 
-            filterExpression = Optional.of(new FilterPredicate(
-                    new FilterPredicate.PathElement(
-                            entityType,
-                            idType,
-                            idField),
-                    Operator.IN,
-                    Collections.singletonList(idVal)));
+            if (!ids.isEmpty()) {
```

Comment:
We need to split the list of IDs into those that already exist and those that are new.  We should build ID filters for the existing and query the data store.  For the IDs that are new, we should get those from the Request scope.

The union of these two sets is what should be returned.

Reply:
Done. See below.